AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -88,122 +93,205 @@ def __init__(self):
+                               "\nThe final and penultimate ranks are reserved for Umpire and Chief of Police")
         return [
-            Label("Rename different ranks of police to fit the game theme"),
-            Label("Ranks affect position on the police page, and can be used to reward/punish/just have a bit of fun"),
-            InputWithDropDown(
-                title="Select rank to rename",
-                identifier=self.html_ids["Rank"],
-                options=[str(i[0]) + ': ' + i[1] for i in list(GENERIC_STATE_DATABASE.arb_state.get(self.identifier).items())]
+            LargeTextEntry(
+                title="Rename Ranks",
+                identifier=self.html_ids["Ranks"],
+                default="\n".join(default_text)
             ),
-            NamedSmallTextbox(
-                title="Enter new name",
-                identifier=self.html_ids["Rank Name"],
-            )
         ]
 
     def answer_set_ranks(self, htmlResponse):
-        rank = htmlResponse[self.html_ids['Rank']]
-        new_name = htmlResponse[self.html_ids['Rank Name']]
-        if new_name:
-            GENERIC_STATE_DATABASE.arb_state[self.identifier][rank[0]] = new_name
-            return [Label(f"[POLICE] Successfully renamed rank {rank} to {new_name}")]
-        else:
-            return [Label(f"[POLICE] Rename failed - no name given")]
+        answer = htmlResponse[self.html_ids['Ranks']].split("\n")
+        if answer[0] != "Add named ranks, rename old/autogenerated ranks, and set default rank." or answer[1] != "The final and penultimate ranks are reserved for Umpire and Chief of Police":
+            return [Label(f"[POLICE] Rename failed - Instructions tampered with")]
+        answer.pop(0)
+        answer.pop(0)  # To remove the instruction lines. Not sure if they are even necessary, but I think they help
+        if len(answer) < 3:
+            # TODO check how many ranks minimum are required from event data.
+            return [Label(f"[POLICE] Rename failed - Must have enough ranks")]
 
-    def on_event_request_create(self) -> List[HTMLComponent]:  # TODO Make a dependent selector, ideally with police filtering
-        return [
-            Dependency(
-                dependentOn="CorePlugin_assassin_pseudonym",
-                htmlComponents=[
-                    AssassinDependentTextEntry(
-                        pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
-                        identifier=self.html_ids["Rank"],
-                        title="Change police rank? Must be a number from 0-9",
-                        default={}
-                    )
-                ]
-            )
-        ]
+        default_counter = 0
+        for idx, ele in enumerate(answer):
+            if ele[-8:] == "=default":
+                default_counter += 1
+                default = idx
+                answer[idx] = ele[:-8]
+        if default_counter == 0:
```

Comment:
In this case, I think it would be sensible to choose the lowest rank and use that as the default.